### ISSUE TITLE
Allow to set default serialization context when building serializer

### ISSUE THREAD

**Issue by nuxwin** [schmittjoh/serializer#528]
Hello ;

Does it is possible to set default context when building the serializer?

Intent: For all our models, we want force serialization of null properties. Right now, we must pass a context object in each of our controllers. For instance:

```php
...
$package = $this->getPackageService()->getPackage($this->getRequest()->query->getInt('package_id'));

$instance = new ApsInstance();
$instance
		->setPackage($package)
		->setSettings($this->getInstanceSettingService()->getSettingObjectsFromMetadataFile($package));

$context = new SerializationContext();
$context->setSerializeNull(true);

$json =  $this->getSerializer()->serialize($instance, 'json', $context);
...
```

Since we want such context for every serialization process, we would be able to do something like this in our Serializer service factory:

```php
namespace iMSCP\Service;

use iMSCP_Registry as Registry;
use JMS\Serializer\SerializerBuilder;
use JMS\Serializer\SerializationContext;
use Zend\ServiceManager\FactoryInterface;
use Zend\ServiceManager\ServiceLocatorInterface;

/**
 * Class SerializerServiceFactory
 * @package iMSCP\Service
 */
class SerializerServiceFactory implements FactoryInterface
{
	/**
	 * {@inheritdoc}
	 */
	public function createService(ServiceLocatorInterface $serviceLocator)
	{
		$serviceLocator->get('Annotation');
		$config = Registry::get('config');

		$defaultContext = new SerializationContext();
		$defaultContext->setSerializeNull(true);

		return SerializerBuilder::create()
			->setDefaultSerializationContext($defaultContext)
			->setCacheDir(CACHE_PATH . '/serializer')
			->setDebug($config['DEVMODE'])
			->build();
	}
}
```

**Comment by alcalyn** [comment]
I had the same problem, then I created a PR: https://github.com/schmittjoh/serializer/pull/541

**Comment by nuxwin** [comment]
@alcalyn

please, don't enforce closure for the factory. Eg, in zf2 service manager, it is recommended to use factory classes which is much better.

**Comment by goetas** [comment]
will be solved by #541 in v1.4
